code/modules/mob/living/simple_animal/abnormality/waw/orange_tree.dm

```diff
@@ -0,0 +1,293 @@
+	var/mob/living/living_pawn = controller.pawn
+	if(!living_pawn.Adjacent(target))
+		return
+	if(living_pawn.next_move > world.time)
+		return
+	living_pawn.changeNext_move(CLICK_CD_MELEE * 5) //No need to be so liberal with it
+	living_pawn.face_atom(target)
+	// AAATTAAACK!
+	living_pawn.UnarmedAttack(target)
+	finish_action(controller, TRUE) //testing line
+
+//Effects
+/obj/effect/temp_visual/dancing_lights
+	icon_state = "dancing_lights"
+	alpha = 0
+	duration = 50
+
+/obj/effect/temp_visual/dancing_lights/Initialize()
+	..()
+	animate(src, alpha = rand(125,200), time = 5)
+	addtimer(CALLBACK(src, .proc/fade_out), 5)
+
+/obj/effect/temp_visual/dancing_lights/proc/fade_out()
+	animate(src, alpha = 0, time = duration-5)
+
+//Disease
+/datum/disease/orange_tree
+	name = "Orange Tree Infection"
+	form = "Infection"
+	max_stages = 5
+	stage_prob = 33
+	infectivity = 100
+	spread_text = "Airborne"
+	disease_flags = CAN_CARRY
+	spread_flags = DISEASE_SPREAD_BLOOD|DISEASE_SPREAD_AIRBORNE
+	cure_text = "Nothing"
+	viable_mobtypes = list(/mob/living/carbon/human)
+	severity = DISEASE_SEVERITY_HARMFUL
+
+/datum/disease/orange_tree/after_add()
+	affected_mob.playsound_local(get_turf(affected_mob), 'sound/abnormalities/orangetree/light3.ogg', 100, 0)
+	RegisterSignal(affected_mob, COMSIG_MOB_APPLY_DAMGE, .proc/DamageCheck)
+
+/datum/disease/orange_tree/cure()
+	UnregisterSignal(affected_mob, COMSIG_MOB_APPLY_DAMGE)
+	affected_mob.visible_message("<span class='nicegreen'>[affected_mob] outwardly appears to have been cleansed of the floating lights.</span>")
+	SanityCheck(TRUE)
+	return ..()
+
+/datum/disease/orange_tree/stage_act()
+	. = ..()
+	if(!.)
+		return
+
+	if(!ishuman(affected_mob))
+		return
+
+	var/mob/living/carbon/human/H = affected_mob
+	H.apply_damage((H.maxSanity * 0.06), WHITE_DAMAGE, null, H.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
+	if(prob(25))
+		var/turf/T = get_turf(H)
+		new /obj/effect/temp_visual/dancing_lights(T)
+
+	if(!H.sanity_lost)
+		if((stage >= max_stages) && (H.sanityhealth >= (H.maxSanity * 0.75)) && prob(H.sanityhealth * 0.25))
+			cure(FALSE)
+		return
+
+	if(!HAS_AI_CONTROLLER_TYPE(H, /datum/ai_controller/insane/murder/orangetree))
+		addtimer(CALLBACK(src, .proc/SanityCheck, FALSE), 1) //Gives sanity time to update
+
+/datum/disease/orange_tree/proc/DamageCheck(mob/living/carbon/human/affected_mob, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/H = affected_mob
+	if(!istype(H))
+		return
+	if(damagetype != RED_DAMAGE)
+		return
+	if(prob(5))
+		cure(FALSE) //very small chance for red damage to cure it
+
+/datum/disease/orange_tree/proc/SanityCheck(var/cured)
+	var/mob/living/carbon/human/H = affected_mob
+	H.visible_message("<span class='danger'>[H] begins mumbling to themselves, seemingly unaware of the rest of the world.</span>")
+	QDEL_NULL(H.ai_controller)
+	H.say("[cured]!")
```

Comment:
This will be `H.say("0!")` sometimes. You should probably move this into `if(cured)`.

Reply:
It's just a testing line.